builder/vsphere/common/step_remove_cdrom.go

```diff
@@ -43,6 +44,30 @@ func (s *StepRemoveCDRom) Run(_ context.Context, state multistep.StateBag) multi
 		}
 	}
 
+	if s.Config.KeepOneCdrom == true {
+		if _, err := vm.FindSATAController(); err == driver.ErrNoSataController {
```

Comment:
I think this should be a switch/case:
- if `sata` was used as the `cdrom_type`, add using `sata`.
- if `ide` was used as the `cdrom_type`, add using `ide`.
- if not provided,  add using `ide` which is the `cdrom_type` default.
-

Reply:
I totally agree. As a total beginner to Go, I tried to keep it simple by arbitrary choosing a SATA config.